### ISSUE TITLE
DDLDatabase fails with MySQL ENUM columns

### ISSUE THREAD

**Issue by anuraaga** [jOOQ/jOOQ#7017]
Parser does not seem to recognize ENUM.

JOOQ 3.10.3

```
SEVERE: An exception occurred while parsing script source : . Please report this error to https://github.com/jOOQ/jOOQ/issues/new
org.jooq.impl.ParserException: C42000_NO_SUBCLASS: Expected tokens: [AUTOINCREMENT, AUTO_INCREMENT, CHECK, DEFAULT, GENERATED, PRIMARY KEY, UNIQUE]: MARY KEY,
  name VARCHAR(255) NOT NULL,
  enabled [*]ENUM('DESTROYED', 'DISABLED', 'ENABLED') DEFAULT 'ENABLED' NOT NULL,
  latitude 
        at org.jooq.impl.ParserImpl$ParserContext.unexpectedToken(ParserImpl.java:5791)
        at org.jooq.impl.ParserImpl.parseQuery(ParserImpl.java:547)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:345)
        at org.jooq.impl.ParserImpl.parse(ParserImpl.java:337)
        at org.jooq.util.ddl.DDLDatabase.create0(DDLDatabase.java:124)
        at org.jooq.util.AbstractDatabase.create(AbstractDatabase.java:237)
        at org.jooq.util.AbstractDatabase.create(AbstractDatabase.java:226)
        at org.jooq.util.AbstractDatabase.getDialect(AbstractDatabase.java:208)
        at org.jooq.util.AbstractGenerator.logDatabaseParameters(AbstractGenerator.java:131)
        at org.jooq.util.JavaGenerator.generate(JavaGenerator.java:271)
        at org.jooq.util.GenerationTool.run(GenerationTool.java:642)
        at org.jooq.util.GenerationTool.generate(GenerationTool.java:203)
        at org.jooq.util.GenerationTool.main(GenerationTool.java:175)
```

**Comment by lukaseder** [comment]
Thank you very much for taking the time to report all these issues! I'll look into these right away.

**Comment by lukaseder** [comment]
This is fixed for 3.11 and will be merged to 3.10.4 next week (#7026). Right now, we can only encode `ENUM` (and `SET`) types as `VARCHAR` types because:

- jOOQ currently cannot internally represent `org.jooq.EnumType` dynamically (see #7025)
- `DDLDatabase` uses an in-memory H2 database internally to run the DDL and reverse engineer it again. H2 doesn't support enums yet (see #5874)

Custom enum types can still be bound by using a converter, though.

**Comment by anuraaga** [comment]
Thanks for the quick response! Out of all the issues I found I assumed this would be hardest, working around with a converter is satisfying enough :)
